graphene_django/tests/test_query.py

```diff
@@ -366,6 +368,61 @@ class Query(graphene.ObjectType):
     assert result.data == expected
 
 
+@pytest.mark.skipif(not DJANGO_FILTER_INSTALLED,
+                    reason="django-filter should be installed")
+def test_should_query_node_filtering_with_distinct_queryset():
+    class FilmType(DjangoObjectType):
+
+        class Meta:
+            model = Film
+            interfaces = (Node, )
+            filter_fields = ('genre',)
+
+    class Query(graphene.ObjectType):
+        films = DjangoConnectionField(FilmType)
+
+        # def resolve_all_reporters_with_berlin_films(self, args, context, info):
+        #    return Reporter.objects.filter(Q(films__film__location__contains="Berlin") | Q(a_choice=1))
+
+        def resolve_films(self, args, context, info):
+            return Film.objects.filter(Q(details__location__contains="Berlin") | Q(genre__in=['ot'])).distinct()
+
+    f = Film.objects.create(
+    )
+    fd = FilmDetails.objects.create(
+        location="Berlin",
+        film=f
+    )
+
+    schema = graphene.Schema(query=Query)
+    query = '''
+        query NodeFilteringQuery {
+            films {
+                edges {
+                    node {
+                        genre
+                    }
+                }
+            }
+        }
+    '''
+
+    expected = {
+        'films': {
+            'edges': [{
+                'node': {
+                    'genre': 'OT'
+                }
+            }]
+        }
+    }
+
+    result = schema.execute(query)
+    assert not result.errors
+    print(result.data)
```

Comment:
Lets be sure to drop this `print` once the test is working.

Reply:
Done

Reply:
the print is still there :)

Reply:
Ah sorry, missed your comment - thanks, will fix now!